### PR TITLE
[202205] Update SAI xgs version to 7.1.32.4-1 and migrate xgs to DMZ

### PR DESCRIPTION
**Pull request by Gfrom2016**
Signed-off-by: zitingguo-ms <zitingguo@microsoft.com>

#### Why I did it
To bring the following MSFT internal fixes, upgrade SAI xgs version to 7.1.32.4-1 and migrate xgs to DMZ repo:
- [bcm sai] enable auto-negotiation support
- [AN/LT] enable Microsoft AN/LT behavior
- Default Not to report Single bit ECC correctable events to avoid the need to set SOC porperties. See Case CS00012218100

#### How I did it
Updated SAI code with the fixes above.

#### How to verify it
Run the SONiC and SAI test with the SAI pipeline.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

File: platform/broadcom/sai.mk [hunk 1]
```diff
@@ -1,8 +1,9 @@
-LIBSAIBCM_XGS_VERSION = 7.1.32.4
+LIBSAIBCM_XGS_VERSION = 7.1.32.4-1
 LIBSAIBCM_DNX_VERSION = 7.1.32.4
-LIBSAIBCM_BRANCH_NAME = REL_7.0_202205
-LIBSAIBCM_XGS_URL_PREFIX = "https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/$(LIBSAIBCM_BRANCH_NAME)/$(LIBSAIBCM_XGS_VERSION)"
-LIBSAIBCM_DNX_URL_PREFIX = "https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/$(LIBSAIBCM_BRANCH_NAME)/$(LIBSAIBCM_DNX_VERSION)"
+LIBSAIBCM_BRANCH_NAME = SAI_7.1.0_GA
```
Comment (richardyu-ms) [inline platform/broadcom/sai.mk:3]: LIBSAIBCM_XGS_BRANCH_NAME

Comment (Gfrom2016) [reply platform/broadcom/sai.mk]: done

**Review by richardyu-ms — APPROVED** [review]

File: platform/broadcom/sai.mk [hunk 2]
```diff
@@ -1,8 +1,9 @@
-LIBSAIBCM_XGS_VERSION = 7.1.32.4
+LIBSAIBCM_XGS_VERSION = 7.1.32.4-1
 LIBSAIBCM_DNX_VERSION = 7.1.32.4
-LIBSAIBCM_BRANCH_NAME = REL_7.0_202205
-LIBSAIBCM_XGS_URL_PREFIX = "https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/$(LIBSAIBCM_BRANCH_NAME)/$(LIBSAIBCM_XGS_VERSION)"
-LIBSAIBCM_DNX_URL_PREFIX = "https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/$(LIBSAIBCM_BRANCH_NAME)/$(LIBSAIBCM_DNX_VERSION)"
+LIBSAIBCM_XGS_BRANCH_NAME = SAI_7.1.0_GA
+LIBSAIBCM_DNX_BRANCH_NAME = REL_7.0_202205
+LIBSAIBCM_XGS_URL_PREFIX = "https://sonicstorage.blob.core.windows.net/public/sai/sai-broadcom/$(LIBSAIBCM_XGS_BRANCH_NAME)/$(LIBSAIBCM_XGS_VERSION)"
```
Comment (richardyu-ms) [inline platform/broadcom/sai.mk:5]: the URL is not right

Comment (Gfrom2016) [reply platform/broadcom/sai.mk]: fix it with adding xgs

**Comment by richardyu-ms** [comment]
please add a description for migrating it to the DMZ repo and re-run the PR test. thanks.

**Pull request closed without merging**